src/data_gradients/datasets/detection/coco_detection_dataset.py

```diff
@@ -0,0 +1,46 @@
+from typing import Union
+
+from data_gradients.datasets.detection.coco_format_detection_dataset import COCOFormatDetectionDataset
+
+
+class COCODetectionDataset(COCOFormatDetectionDataset):
+    """Coco Detection Dataset expects the exact same annotation files and dataset structure os the original Coco dataset.
+
+    #### Expected folder structure
+    The dataset folder structure should
+
+    Example:
+    ```
+    dataset_root/
+        ├── images/
+        │   ├── train2017/
+        │   ├── val2017/
+        │   └── ...
+        └── annotations/
+            ├── instances_train2017.json
+            ├── instances_val2017.json
+            └── ...
+    ```
+
+    #### Instantiation
+    To instantiate a dataset object for training data of the year 2017, use the following code:
+
+    ```python
+    from data_gradients.datasets.detection import COCODetectionDataset
+
+    train_set = COCODetectionDataset(root_dir="<path/to/dataset_root>", split="train", year=2017)
+    val_set = COCODetectionDataset(root_dir="<path/to/dataset_root>", split="val", year=2017)
+    ```
+    """
+
+    def __init__(self, root_dir: str, split: str, year: Union[int, str] = 2017):
```

Comment:
Do we really need a second class here?

Reply:
Ok, I'll elaborate.
What is the added value of this class and does it outweighs the added support cost of the new class, whose only purpose is to support multiple versions of the official COCO datasets?

Reply:
PRO
- Ease of use: the user does not need to dig into the dataset folders to know its inner structure.
- (Consequence of previous) Less likely for users to make mistakes, since we handle path mapping

CON
- If someone uses this class on a version of COCO that is structured differently, then the errors will not be explicit (because we assume it is indeed COCO, and structured like COCO)
- (and of course) adding more code

Do you see other points ?
I think the pros>cons, so I would tend to think it's better to keep, but I'm good with either choice. What do you think?

Reply:
As I said - I don't see a big advantage in two classes, but that's not a deal breaker to me